ExampleApp/ExampleApp/Common/Services/RxBluetoothKitService.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+import RxBluetoothKit
+import RxSwift
+import RxCocoa
+
+// RxBluetoothKitService is a class encapsulating logic of most operations you might want to perform
+// on a CentralManager object. Here you can see an example usage of such features as scanning for Peripherals,
+//
+
+class RxBluetoothKitService {
+
+    typealias Disconnection = (Peripheral, DisconnectionReason?)
+
+    // MARK: - Public outputs
+
+    var scanningOutput: Observable<ScannedPeripheral> {
+        return scanningSubject.share(replay: 1, scope: .forever).asObservable()
+    }
+
+    var servicesOutput: Observable<[Service]> {
+        return servicesSubject.asObservable()
+    }
+
+    var disconnectionReasonOutput: Observable<Disconnection> {
+        return disconnectionSubject.asObservable()
+    }
+
+    var errorOutput: Observable<Error> {
+        return errorSubject.asObservable()
+    }
+
+    // MARK: - Private subjects
+
+    private let scanningSubject: PublishSubject<ScannedPeripheral> = PublishSubject()
+
+    private let servicesSubject: PublishSubject<[Service]> = PublishSubject()
+
+    private let disconnectionSubject: PublishSubject<Disconnection> = PublishSubject()
+
+    private let errorSubject: PublishSubject<Error> = PublishSubject()
+
+    // MARK: - Private fields
+    private let centralManager: CentralManager = CentralManager(queue: .main)
+
+    private let scheduler: ConcurrentDispatchQueueScheduler
+
+    private let disposeBag = DisposeBag()
+
+    private var connectedPeripherals: [Peripheral] = []
+
+    private var scanningDisposable: Disposable!
+
+    // MARK: - Initialization
+    init() {
+        let timerQueue = DispatchQueue(label: Constant.Strings.defaultDispatchQueueLabel)
+        scheduler = ConcurrentDispatchQueueScheduler(queue: timerQueue)
+    }
+
+
+    // MARK: - Scanning for peripherals
+
+    // You start from observing state of your CentralManager object. Within RxBluetoothKit v.5.0, it is crucial
+    // that you use .startWith(:_) operator, and pass the initial state of your CentralManager with
+    // centralManager.state.
+    func startScanning() {
+        scanningDisposable = centralManager.observeState()
+                .startWith(centralManager.state)
+                .filter {
+                    $0 == .poweredOn
+                }
+                .subscribeOn(MainScheduler.instance)
+                .timeout(4.0, scheduler: scheduler)
+                .flatMap { [unowned self] _ -> Observable<ScannedPeripheral> in
```

Comment:
Are you sure, that we can use `unowned` here?

Reply:
Done